db/migrate/20230516072511_add_created_manually_flag_to_customer.rb

```diff
@@ -0,0 +1,5 @@
+class AddCreatedManuallyFlagToCustomer < ActiveRecord::Migration[7.0]
+  def change
+    add_column :customers, :created_manually, :boolean, default: false
```

Comment:
Yes, the customers that were created manually and **don't have at least one completed/resumed order** will disappear.

There's no way to distinguish between the two.
 
I think a subset of them are the customers who don't have any orders associated with them (regardless of the state of the order). Maybe we can add a migration to update their flag?

if we add that migration, we'll only lose the customers **created manually**,  *have at least incomplete order*, and **don't have any completed order** (the three conditions must be verified).  

what do you think?

Reply:
That's a good idea. We can look for customers without any orders and it's safe to assume that they were added manually. We cant distinguish the others... hm, what about the created_at column of order and customer? If a customer record was created more than a minute before the order record, it must have been created manually.

Uh, another thing is tags. If enterprises assigned tags to customers then they probably want to continue to see them. That makes me wonder what happens if an enterprise tries to create a customer which already exists but is invisible. It probably fails. And there's no way of tagging such a customer. We need a way to "activate" customers or we just switch them to manually created if we detect this.

Reply:
> what happens if an enterprise tries to create a customer which already exists but is invisible. It probably fails. And there's no way of tagging such a customer.

^ This seems really important. We probably should provide a filter option so that users can choose to show/hide based on this field (I think this would apply to the admin interface and API).

Even then, it may not be clear when you get the error "customer already exists" but you can't see them. So we might need to add a hint in the "customer already exists" message (assuming there is one).